### PR TITLE
Major refactor.

### PR DESCRIPTION
**Pull request by AntonAFA**
All the parameters moved to standalone object - KavaParamsModel. It will be updated by KavaAnalyticsPlugin with most relevant data when needed KavaAnalyticsPlugin will request for the up to date data and receive the Map with all of the parameters that are relevant for event type.

+ added support for playTimeSum.
+ some small bug fixes.

### CODE REVIEW

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    private Map<Long, Long> averageTrackPlaybackDuration = new HashMap<>();
+
+    /**
+     * Calculate average bitrate for the entire media session.
+     * @param totalPlaytimeSum - total amount of the player being in active playback mode.
+     * @return - average bitrate.
+     */
+    long getAverageBitrate(long totalPlaytimeSum) {
+        if (!shouldCount) {
+            log.w("Should not try to get average bitrate while player is paused");
```
Comment (giladna) [inline kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: return is missing?

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        updateBitratePlayTime();
+        Iterator iterator = averageTrackPlaybackDuration.entrySet().iterator();
+        long averageBitrate = 0;
+        while (iterator.hasNext()) {
+            Map.Entry pair = (Map.Entry)iterator.next();
```
Comment (giladna) [inline kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: if you use Generics here you do not need casting

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            long bitrate = (long) pair.getKey();
+            long playTime = (long) pair.getValue();
+            averageBitrate += (bitrate * playTime) / totalPlaytimeSum;
+
```
Comment (giladna) [inline kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: make sure totalPlaytimeSum != 0

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+            totalPlayedTime += playedTime;
+            averageTrackPlaybackDuration.put(currentTrackBitrate, totalPlayedTime);
+        }
+
+        currentTrackStartTimestamp = currentTimeStamp;
+
+    }
+
+    void resumeCounting() {
+        currentTrackStartTimestamp = System.currentTimeMillis();
```
Comment (giladna) [inline kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: what happens when going to bg?

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java (hunk 3 above)
Comment (AntonAFA) [reply kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: totalPlaytimeSum can`t be 0. getAverage is requested from VIEW event which means that there was at least 10 seconds of active playback.

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java (hunk 2 above)
Comment (AntonAFA) [reply kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: Done!

File: kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java (hunk 1 above)
Comment (AntonAFA) [reply kavaplugin/src/main/java/com/kaltura/playkit/plugins/kava/AverageBitrateCounter.java]: no need to return. Instead removed.

**Pull request merged**